src/cargo/core/workspace.rs

```diff
@@ -1215,71 +1216,126 @@ impl<'cfg> Workspace<'cfg> {
                 } => panic!("unexpected dep: syntax {}", feature),
                 FeatureValue::DepFeature {
                     dep_name,
-                    dep_feature,
+                    dep_feature: _,
                     dep_prefix: _,
                     weak: _,
                 } => {
-                    // I think weak can be ignored here.
-                    // * With `--features member?/feat -p member`, the ? doesn't
-                    //   really mean anything (either the member is built or it isn't).
-                    // * With `--features nonmember?/feat`, cwd_features will
-                    //   handle processing it correctly.
+                    // Check if `dep_name` is member of the workspace or package.
+                    // Weak can be ignored for this moment.
                     let is_member = self.members().any(|member| member.name() == *dep_name);
                     if is_member && specs.iter().any(|spec| spec.name() == *dep_name) {
                         member_specific_features
                             .entry(*dep_name)
                             .or_default()
-                            .insert(FeatureValue::Feature(*dep_feature));
+                            .insert(feature.clone());
```

Comment:
Would it be possible to leave this line unchanged?  Then the code below wouldn't need to change (the call to `map` which calls `FeatureValue::new`), since it should be able to pass the `FeatureValue::Feature` through as-is.

Reply:
@ehuss We will lose ability to print features as they are.

Reply:
Ah, I see.

Reply:
@ehuss we will lose nice error message, when assert fails

Reply:
I think that should be fine.  Also, as-is, the assert doesn't display anything other than the map is not empty, but doesn't display its contents.